In [5]:
import tensorflow as tf
tf.compat.v1.disable_eager_execution()

w = tf.Variable([0.2], dtype=tf.float32)   # initial weight
b = tf.Variable([-0.2], dtype=tf.float32)  # initial bias

x = tf.compat.v1.placeholder(tf.float32)   # input
y = tf.compat.v1.placeholder(tf.float32)   # actual output

linear_model = w * x + b

In [6]:
init = tf.compat.v1.global_variables_initializer()  # initializing variables

s = tf.compat.v1.Session()  # setting up session for computation

s.run(init)

print(s.run(w))
print(s.run(b))

[0.2]
[-0.2]


# Lets calculate the error after 1st epoch

In [7]:
squared_deltas=tf.square(linear_model -y)  # used to square the error
loss = tf.reduce_sum(squared_deltas) # reduce_sum is used to sum all the squared errors and give us rmse


print("loss or Error is ")
print(s.run(loss, {x:[1,2,3,4], y:[0,-1,-2,-3]}))   #given input and output values here because we have used placeholder


loss or Error is 
20.16


# error is 20.16 . So lets update the weights and check error again

In [9]:
# now we want to reduce the RMSE/loss, so we will use gradient descent optimizer

optimizer = tf.compat.v1.train.GradientDescentOptimizer(0.01)  # 0.01 is learning rate
train = optimizer.minimize(loss)

s.run(init)

for i in range(1000):  # 1000 training epochs
    s.run(train, feed_dict={x: [1, 2, 3, 4], y: [0, -1, -2, -3]})

In [10]:
print(s.run([w,b])) # we will get values for weight and bias


[array([-0.99999714], dtype=float32), array([0.99999154], dtype=float32)]


In [11]:
s.run(linear_model, {x:[1,2,3,4], y:[0,-1,-2,-3]})


array([-5.6028366e-06, -1.0000027e+00, -1.9999999e+00, -2.9999971e+00],
      dtype=float32)

these values are near to the desired output

In [12]:
s.run(loss, {x:[1,2,3,4], y:[0,-1,-2,-3]})



np.float32(4.7108983e-11)

# RMSE after gradient descent optimizer is 4.7108983e-11. Earlier we were getting 20.16